lib/workers/repository/update/branch/index.ts

```diff
@@ -553,26 +553,35 @@ export async function processBranch(
     await setStability(config);
     await setConfidence(config);
 
-    // break if we pushed a new commit because status check are pretty sure pending but maybe not reported yet
+    const notImmediate = commitSha && config.prCreation !== 'immediate';
+
+    // new commit means status check are pretty sure pending but maybe not reported yet
+    // if PR has not been created + new commit + prCreation !== immediate skip
     // but do not break when there are artifact errors
     if (
+      notImmediate &&
+      !branchPr &&
       !config.artifactErrors?.length &&
-      !userRebaseRequested &&
-      commitSha &&
-      config.prCreation !== 'immediate'
+      !userRebaseRequested
     ) {
-      logger.debug(`Branch status pending, current sha: ${commitSha}`);
+      logger.debug(`Branch status pending, current sha: ${commitSha!}`);
       return {
         branchExists: true,
         updatesVerified,
-        prNo: branchPr?.number,
         result: BranchResult.Pending,
         commitSha,
       };
     }
 
     // Try to automerge branch and finish if successful, but only if branch already existed before this run
-    if (branchExists || config.ignoreTests) {
+    // skip if we have a non-immediate pr and there is an existing PR,
+    // we want to update the PR and skip the Auto merge since status checks aren't done yet
+    if (
+      !notImmediate &&
```

Comment:
does the ignoreTests relate to the commitSha && config.prCreation !== 'immediate';?
if i keep same funcitonality in a flow where
prCreation !== immediate and ignoreTests=true and there's a new commit, 
the code used to break.
but if we are ignoring tests, then why should we break and skip automerge? we are ignoring tests anyway

i'm doubting this flow

Reply:
or does prCreation relate to ALL checks including tests?

Reply:
no, we should never try to automerge when we've done a fresh commit

btw: here's another automerge condition (pr-automerge)

https://github.com/renovatebot/renovate/blob/76b536e23a1b25999c2ffcf029bcb4e786dd4753/lib/workers/repository/update/branch/index.ts#L812-L821

Reply:
i dont think i need to change anything here since inside the check auto merge there's always a check for branch status, so a new commit or not a new commit, the branch status check will give the indication if we should continue or not

Reply:
so you want to skip both in case pr !== immediate? hmmmm this is more complicated than i thought

Reply:
no, the status checks can be missing and renovate can have added a successful stabillity check, so renovate would strait automerge. so we should never try to automerge when we've done a new commit.

Reply:
![image](https://user-images.githubusercontent.com/42116482/201112937-7d23cf49-ebd3-44f7-a423-bf659aa85432.png)
its not defined in there :D

Reply:
ah, ok, i've done it but now i have a problem in 1 of the tests see my comment blow
https://github.com/renovatebot/renovate/pull/18839#issuecomment-1310425195